### PR TITLE
Add some dist-training robust cases into fluid benchmark test

### LINKED ISSUE #11206: Add distributed training robust cases into fluid benchmark test
the PaddlePaddle robust feature combination includes:

Parallel Executor | slice_var_up | sync_mode | Place | Parameters Update Method | Trainer Count | model | LearningRate Decay | Regularization | Gradient Clipping | Error Clipping
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
True | True | True | GPU | pserver | 2 | Mnist | True | L1 | True | True
False | False | False | CPU | nccl2 |   |   | False | L2 | False | False
  |   |   |   |   |   |   |   | None |   |  

I construct the test-case-pairs with [AllPair](https://pypi.org/project/AllPairs/) as following:

```
import os
import metacomm.combinatorics.all_pairs2
all_pairs = metacomm.combinatorics.all_pairs2.all_pairs2

parameters = [ [ "--gpus 2", "--gpus 1" ] # If use ParallelExecutor or NOT
             , [ "", "--no_split_var"] # If split variables into blocks or NOT
             , [ "", "--async_mode" ] # if ASGD
             , [ "--device GPU", "--device CPU" ] # if train with GPU
             , [ "--model mnist" ] # model
             , [ "--update_method pserver", "--update_method nccl2" ] # parameters update method
             , [ "--learning_rate_decay_method exponential", "--learning_rate_decay_method natural_exp" ] # learning rate decay
             , [ "--weight_decay_regularizer_method L1", "--weight_decay_regularizer_method L2" ]
             , [ "--gradient_clip_method Norm", "--gradient_clip_method GlobalNorm" ]
             , [ "--error_clip_method Value", "" ]
             ]

pairwise = all_pairs(parameters)

print "PAIRWISE:"
for i, v in enumerate(pairwise):
    cmd = "python fluid_benchmark.py"
    for arg in v:
        cmd += ' ' + arg
    print "%i:\t%s" % (i, cmd)
```

### PR DESCRIPTION
**Pull request by velconia**
Close #11206

### CODE REVIEW

File: benchmark/fluid/models/machine_translation.py [hunk 1]
```diff
@@ -26,6 +26,10 @@
 import paddle.fluid.core as core
 import paddle.fluid.framework as framework
 from paddle.fluid.executor import Executor
+from models.model_base import get_decay_learning_rate
```
Comment (typhoonzero) [inline benchmark/fluid/models/machine_translation.py]: model_base is not uploaded?

Comment (velconia) [reply benchmark/fluid/models/machine_translation.py]: Thanks for review, I added the benchmark/fluid/models/model_base.py file in next commit

File: benchmark/fluid/models/machine_translation.py [hunk 2]
```diff
@@ -211,12 +217,24 @@ def get_model(args):
...
-        max_length=max_length)
+        max_length=max_length,
+        args=args)
 
     # clone from default main program
     inference_program = fluid.default_main_program().clone()
 
-    optimizer = fluid.optimizer.Adam(learning_rate=args.learning_rate)
+    # set gradient clip
+    set_gradient_clip(args.gradient_clip_method, args.gradient_clip_norm)
```
Comment (typhoonzero) [inline benchmark/fluid/models/machine_translation.py]: Is there a way that we can disable these settings if the args is empty?

Comment (velconia) [reply benchmark/fluid/models/machine_translation.py]: if clip_method in args is None, these settings will be disabled, and if user do NOT specify the args ```--gradient_clip_method```, the args will be None in the case of default.

the code was like below
```
def set_gradient_clip(clip_method, clip_norm=1.):
    if not clip_method:
        return None
```

**Comment by typhoonzero** [comment]
I'm currently thinking, we can test all these cases using unit test and not CE. Run e2e tests with CE may spend alot of time

**Comment by velconia** [comment]
Actually, test cases have cover the most part of these features, however, what we need is:

1. Running an program in distributed  environment, e.g., run learning decay with parallel executor in 2 parameter servers and 2 trainers which unit tests could not cover.
2. These tests does not run to end actually, we just run few iterations each time, which will not cost much time.

**Comment by velconia** [comment]
So I guess, put these features in fluid benchmark and add about 6-7 cases in ce is a good choice, which will cost around 30s each case

File: benchmark/fluid/args.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        help='Gradient clipping norm, 1. for default')
+    #  args related to error clipping
+    parser.add_argument(
+        '--error_clip_method',
+        type=str,
+        default=None,
+        choices=[],
+        help='Error clipping method, not allowed yet')
+    parser.add_argument(
+        '--error_clip_min',
```
Comment (typhoonzero) [inline benchmark/fluid/args.py]: Can we remove clipping and other optimization configures in argument, It might be clean if we leave these settings to model configs, Thanks!

Comment (velconia) [reply benchmark/fluid/args.py]: done

File: python/paddle/fluid/tests/unittests/test_listen_and_serv_op.py [hunk 4]
```diff
@@ -71,16 +71,17 @@ def _start_pserver(self, use_cuda, sync_mode):
...
-            time.sleep(0.5)
+            time.sleep(sleep_time)
             try:
                 # the listen_and_serv_op would touch a file which contains the listen port
                 # on the /tmp directory until it was ready to process all the RPC call.
                 os.stat("/tmp/paddle.%d.port" % pid)
                 return
             except os.error:
-                retry_times -= 1
+                retry_times -= sleep_time
```
Comment (typhoonzero) [inline python/paddle/fluid/tests/unittests/test_listen_and_serv_op.py]: This change seems is not for current PR? and `retry_times` seems is the total count for trying not the time.

Comment (velconia) [reply python/paddle/fluid/tests/unittests/test_listen_and_serv_op.py]: I changed the name ```retry_times``` to ```start_left_time``` to indicate that this is the left time for pserver starting, and this change is for passing the CI

**Comment by typhoonzero** [comment]
Ref https://github.com/PaddlePaddle/Paddle/issues/11213 for adding unit tests
